client/asset/btc/btc.go

```diff
@@ -772,8 +773,9 @@ func (btc *ExchangeWallet) maxOrder(lotSize, feeSuggestion uint64, nfo *dex.Asse
 	if err != nil {
 		return nil, nil, fmt.Errorf("error parsing unspent outputs: %w", err)
 	}
-	// Start by attempting max lots with no fees.
-	lots := avail / lotSize
+	// Start by attempting max lots with a basic fee.
+	basicFee := nfo.SwapSize * nfo.MaxFeeRate
+	lots := avail / (lotSize + basicFee)
```

Comment:
What is this `basicFee` adjustment for? This seems wrong. `estimateSwap` will use the feeSuggestion internally.

Reply:
I think it is returning before using the feeSuggestion at `tryFund`. 

The problem I am seeing is that the below estimateSwap will run an extreme number of times for large numbers of swaps if not including a fee. Please try on master with a small lot size and large amount of asset (dcr) and I believe your client will hang. Does it not?

Enter values similar to these on simnet.

![image](https://user-images.githubusercontent.com/34998433/121830002-03f44900-ccff-11eb-8e65-42b6fd270f4a.png)

Reply:
Yes, confirmed.  It's a problem.  Even for more reasonable values, the loop (`for lots > 0 {`) iterates excessively and we don't always realize it.

When the starting `lots` is huge, as Joe described, it has to iterate a lot more to hit on the appropriate fees.  Note than the btc ExchangeWallet.maxOrder method, `avail` and `lotSize` are sats, so when you type 0.0001 as the rate in a buy order in a DCR-BTC market with 10 DCR lot size, it starts the method with `lotSize` of 100000.  With the default simnet BTC wallet, 44000000000 sats available, and that's 440000 starting lots.  It iterates down to 381611 lots, one lot less at a time.

It is correct that `estimateSwap` returns because fund/tryfund has an error with the provided number of lots, so the loop continues with one less lot, but it's often needing to iterate thousands of times and it effectively hangs.

I think we need to start with some fees as Joe is suggesting or rethink the maxOrder algo.  `basicFee := nfo.SwapSize * nfo.MaxFeeRate` does seem correct here.

Reply:
With the proposed change:

```
avail: 44000000000  lotSize: 100000
start btc lots: 381613
final btc lots:  381611
```

So it only has to iterate twice to get the highest possible lots.